hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
When a proxy responds with 200 to a CONNECT request, it doesn't include a `proxy-connection` header at all (at least I haven't seen that).
But when it responds with >=400, it does include a `proxy-connection: close` header, which should be respected here.

So I think there's no need to make any exception for a CONNECT request here.